### PR TITLE
Sundry fixes

### PR DESCRIPTION
**Pull request by mruwnik**
* fix the subcategories bug
* don't show deleted articles
* allow `/question/<pageid>` redirects

image

### CODE REVIEW

**Comment by mruwnik** [comment]
why are the tests failing... :/

**Comment by mruwnik** [comment]
aha. Npm is having problems...
image

File: app/routes/questions.$questionId.$.tsx [hunk 1]
```diff
@@ -163,6 +179,8 @@ export default function RenderArticle() {
                 return (
                   <Error error={{statusText: 'Could not fetch question', status: 'emptyArticle'}} />
                 )
+              } else if (resolvedQuestion.data.status === QuestionStatus.TO_DELETE) {
```
[the remark this answers is not in the document]
Comment (LeMurphant) [reply app/routes/questions.$questionId.$.tsx:182]: **Duplicate**
They are used essentially in the same way, they can both have the same error msg.

**Review by Aprillion — DISMISSED** [review]

File: app/routes/questions.toc.ts [hunk 2]
```diff
@@ -79,10 +80,16 @@ export const loadToC = async (request: any): Promise<LoaderResp> => {
...
+
   return {
     data: data
       .filter(canBeShown)
-      .filter(({tags}) => tags?.includes(INTRODUCTORY) || tags?.includes(ADVANCED))
+      .filter(
+        ({tags, status}) =>
+          tags?.includes(INTRODUCTORY) ||
+          tags?.includes(ADVANCED) ||
+          status === QuestionStatus.SUBSECTION
```
[the remark this answers is not in the document]
Comment (LeMurphant) [reply app/routes/questions.toc.ts:91]: Subsections are actually never tagged as either introductory or advanced, only sections are tagged as such. Still, I think the code makes sense. Did you mean that weird things will happen if a subsection _is_ tagged as introductory or advanced?

File: app/routes/questions.$questionId.$.tsx [hunk 3]
```diff
@@ -21,6 +26,17 @@ import {ArticlesNavManualList} from '~/components/ArticlesNav/ArticleNavManualLi
 export const LINK_WITHOUT_DETAILS_CLS = 'link-without-details'
 
 export const loader = async ({request, params}: LoaderFunctionArgs) => {
+  const {data: redirects} = await loadRedirects(request)
+
+  // Check if we need to redirect to a new path
```
[the remark this answers is not in the document]
Comment (LeMurphant) [reply app/routes/questions.$questionId.$.tsx:31]: https://aisafety.info/questions/8TJV/How-can-I-help
now renders as "deleted", but the redirection is not working (see https://coda.io/d/AI-Safety-Info_dfau7sl2hmG/Redirects_suCc9XjM#Redirects_tuDQLOm3)
This is a local improvement over the previous situation, but I'd still rather redirections take priority over deleted articles

File: app/routes/questions.toc.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mruwnik) [reply app/routes/questions.toc.ts:91]: I don't know if they *will* happen, but I'm not certain that they won't...

**Pull request merged**